docs/specification/draft/basic/utilities/tasks.mdx

```diff
@@ -0,0 +1,803 @@
+    C->>LLM: Request completion
+    LLM->>C: Return completion
+
+    Note over C,U: Client may present result to user
+    C->>U: Review result
+    U->>C: Approve result
+
+    Note over S,C: Server polls and discovers completion
+    S->>C: tasks/get (request-789)
+    C->>S: completed
+
+    Note over S,C: Server retrieves result
+    S->>C: tasks/result (request-789)
+    C->>S: Result content
+
+    Note over S: Server continues processing
+
+    Note over C: Results retained for keepAlive period
+```
+
+### Task Cancellation Flow
+
+```mermaid
+sequenceDiagram
+    participant C as Client (Requestor)
+    participant S as Server (Receiver)
+
+    Note over C,S: 1. Task Creation
+    C->>S: tools/call (request ID: 42, task-123)
+    S--)C: notifications/tasks/created
+
+    Note over C,S: 2. Task Processing
+    C->>S: tasks/get (task-123)
+    S->>C: working
+
+    Note over C,S: 3. Client Cancellation
+    Note over C: User requests cancellation
+    C--)S: notifications/cancelled (requestId: 42)
+
+    Note over S: Server processes cancellation
+    Note over S: Task moves to cancelled status
+
+    Note over C,S: 4. Confirmation Polling
+    C->>S: tasks/get (task-123)
+    S->>C: cancelled
+
+    Note over C: Client confirms cancellation succeeded
+
+    Note over S: After keepAlive period, task cleaned up
+```
+
+### Task Deletion Flow
+
+```mermaid
+sequenceDiagram
+    participant C as Client (Requestor)
+    participant S as Server (Receiver)
+
+    Note over C,S: 1. Task Creation and Completion
+    C->>S: tools/call (task-123)
+    S--)C: notifications/tasks/created
+    Note over S: Task processing...
+    C->>S: tasks/get (task-123)
+    S->>C: completed
+
+    Note over C,S: 2. Result Retrieval
+    C->>S: tasks/result (task-123)
+    S->>C: Result content
+
+    Note over C,S: 3. Explicit Deletion
+    Note over C: Client decides to clean up task
+    C->>S: tasks/delete (task-123)
+    S->>C: Success (empty result)
+
+    Note over S: Task and results immediately deleted
+
+    Note over C,S: 4. Verification (optional)
+    C->>S: tasks/get (task-123)
+    S->>C: Error: Task not found
+```
+
+## Data Types
+
+### Task
+
+A task represents the execution state of a request. The task metadata includes:
+
+- `taskId`: Unique identifier for the task
+- `keepAlive`: Time in milliseconds that results will be kept available after completion
+- `pollInterval`: Suggested time in milliseconds between status checks
+- `status`: Current state of the task execution
+
+### Task Status
+
+Tasks can be in one of the following states:
+
+- `submitted`: The request has been received and queued for execution
+- `working`: The request is currently being processed
+- `input_required`: The task lifecycle itself encountered an error, unrelated to the associated request logic
+- `completed`: The request completed successfully and results are available
+- `failed`: The task lifecycle itself encountered an error
```

Comment:
I think it would be more consistent to make "failed" reflect the state of the request rather than the "lifecycle". This was originally about the request but was changed to exclude the case of a tool call whose execution failed (`isError = true`). But now there is the same issue with `completed`. When a tool call execution fails, is the task `completed`? I.e. did the "request complete successfully"?

I think a better design is to let "completed" and "failed" be about the request. Then allow `tasks/result` to be issued for tasks in the failed state. I think this is consistent with the way results work for tool calls today and will be easier for developers to understand and implement.